platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/IssDetector.java

```diff
@@ -194,10 +210,44 @@ public void signalEndOfPreconsensusReplay(@Nullable final Object ignored) {
         }
 
         final long roundWeight = addressBook.getTotalWeight();
-        previousRound = round;
+        previousRound = roundNumber;
 
-        roundData.put(round, new RoundHashValidator(round, roundWeight, issMetrics));
-        return listOrNull(removedRounds.stream().map(this::handleRemovedRound).toList());
+        roundData.put(roundNumber, new RoundHashValidator(roundNumber, roundWeight, issMetrics));
+
+        return removedRounds.stream()
+                .map(this::handleRemovedRound)
+                .filter(Objects::nonNull)
+                .toList();
+    }
+
+    /**
+     * Called when a round has been completed.
+     * <p>
+     * Expects the contained state to have been reserved by the caller for this method. This method will release the
+     * state reservation when it is done with it.
+     *
+     * @param stateAndRound the round and state to be handled
+     * @return a list of ISS notifications, or null if no ISS occurred
+     */
+    public @Nullable List<IssNotification> handleStateAndRound(@NonNull final StateAndRound stateAndRound) {
+        try (final ReservedSignedState state = stateAndRound.reservedSignedState(); ) {
+            final long roundNumber = stateAndRound.round().getRoundNum();
+            if (isRoundToIgnore(roundNumber)) {
+                return null;
+            }
```

Comment:
I don't think the logic for ignored rounds is correct. If we want to ignore ISS detection during a specific round, it's still ok to collect signature transactions that reach consensus during that round. Also, we should still be shifting the window even if we want to ignore the round.

Would it be possible to implement this so that we do everything like normal when we are given a round we want to ignore except that we throw away ISS notifications for an ignored round? (regardless of the round during which the ISS notification was raised)